Update "Web embedding" documentation. (#10977)
flutter/website: src/content/platform-integration/web/web-content-in-flutter.md

* `HtmlElementView.fromTagName`.
* `HtmlElementView` and `registerViewFactory`.

### `HtmlElementView.fromTagName`

The [`HtmlElementView.fromTagName` constructor][] creates an HTML Element from
its `tagName`, and provides an `onElementCreated` method to configure that
element before it's injected into the DOM:

```dart
// Create a `video` tag, and set its `src` and some `style` properties...
HtmlElementView.fromTag('video', onElementCreated: (Object video) {
  video as web.HTMLVideoElement;
  video.src = 'https://interactive-examples.mdn.mozilla.net/media/cc0-videos/flower.mp4';
  video.style.width = '100%';
  video.style.height = '100%';
  // other customizations to the element...
});
```

Check out [`package:web`][] to learn more about the [`HTMLVideoElement` class].
It's the recommended way to interact with DOM APIs.

Learn more about the `Object` to `web.HTMLVideoElement` cast in Dart's
[JS Interoperability][] documentation.

[`HtmlElementView.fromTagName` constructor]: {{site.api}}/flutter/widgets/HtmlElementView/HtmlElementView.fromTagName.html
[`HTMLVideoElement` class]: {{site.pub}}/documentation/web/latest/web/HTMLVideoElement-extension-type.html

> [`package:web`]: {{site.pub}}/packages/web

Author: Good catch!!

### `HtmlElementView` and `registerViewFactory`

If you need more control over generating the HTML code you inject, you can use
the primitives that Flutter uses to implement the `fromTagName` constructor. In
this scenario, register your own HTML Element factory for each type of HTML
content that needs to be added to your app.

> The resulting code is more verbose, and it has two steps per platform view type:

Reviewer: ```suggestion
The resulting code is more verbose, and has two steps per platform view type:
```

1. Register the HTML Element Factory using
`platformViewRegistry.registerViewFactory` provided by `dart:ui_web.`  
2. Place the widget with the desired `viewType`  with
`HtmlElementView('viewType')` in your app's widget tree.

> See more details about this approach in the Flutter documentation for the
> [`HtmlElementView` widget][].

Reviewer: ```suggestion
For more details about this approach, check out
[`HtmlElementView` widget][] docs.
```
